pkg/provider/azure_routes.go

```diff
@@ -105,11 +105,6 @@ func (d *delayedRouteUpdater) updateRoutes() {
 	d.lock.Lock()
 	defer d.lock.Unlock()
 
-	// No need to do any updating.
-	if len(d.routesToUpdate) == 0 {
```

Comment:
Could we keep this check so that the outdated routes would only be cleanup when there're routes need to be deleted or added?

Reply:
Yes, since the conflict issue happens only if the route table needs to be updated.

Reply:
done